src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -431,30 +433,62 @@ private RestAzureNS.AzureOperationResponse<ProtectionPolicyResource> CreateorMod
                                 
                 if (retentionPolicy != null)
                 {
-                    AzureWorkloadProviderHelper.ValidateLongTermRetentionPolicy(retentionPolicy, BackupManagementType.AzureStorage, ScheduleRunFrequency);
-                    ((AzureFileSharePolicy)policy).RetentionPolicy = retentionPolicy;
-                    Logger.Instance.WriteDebug("Validation of Retention policy is successful");
-                }               
-
-                // copy the backupSchedule time to retentionPolicy after converting to UTC
-                AzureWorkloadProviderHelper.CopyScheduleTimeToRetentionTimes(
-                    (CmdletModel.LongTermRetentionPolicy)((AzureFileSharePolicy)policy).RetentionPolicy,
-                    (CmdletModel.SimpleSchedulePolicy)((AzureFileSharePolicy)policy).SchedulePolicy);
-                Logger.Instance.WriteDebug("Copy of RetentionTime from with SchedulePolicy to RetentionPolicy is successful");
+                    if (ProviderData.ContainsKey(PolicyParams.RetentionPolicy) && ProviderData[PolicyParams.RetentionPolicy] != null && ProviderData[PolicyParams.RetentionPolicy].GetType() == typeof(CmdletModel.VaultRetentionPolicy))
```

Comment:
ProviderData.ContainsKey and ProviderData[key] here and below appeared multiple times. It could cause potential issue when running with multi-threading. Please consider using TryGet method instead if possible.

Reply:
@vidai-msft Updated code to use TryGet method